interceptor/pkg/interceptor/pdinterceptor.go

```diff
@@ -90,15 +90,36 @@ func (pdi *PagerDutyInterceptor) executeInterceptor(r *http.Request) ([]byte, er
 	}
 	r.Body = io.NopCloser(bytes.NewReader(body.Bytes()))
 
+	var wrapper struct {
+		Body   string              `json:"body"`
+		Header map[string][]string `json:"header"`
+	}
+	if err := json.Unmarshal(body.Bytes(), &wrapper); err != nil {
+		return nil, badRequest(fmt.Errorf("failed to parse body as JSON: %w", err))
+	}
+
+	extractedRequest, err := http.NewRequestWithContext(ctx, r.Method, r.URL.String(), bytes.NewReader([]byte(wrapper.Body)))
+	if err != nil {
+		return nil, internal(fmt.Errorf("failed to create new request: %w", err))
```

Comment:
thought: while this error message is accurate, i can imagine it might be confusing as there is no request happening, maybe something along the lines of, malformed request body/header ( in unwrapped request ) ?